src/Datasource/EntityTrait.php

```diff
@@ -787,6 +787,38 @@ public function getDirty()
     }
 
     /**
+     * return value if property exists otherwise throw Exception
+     *
+     * @param string $property The field to get value for.
+     * @return mixed
+     */
+    public function getOrFail($property)
+    {
+        $value = $this->get($property);
+        if (!$this->propertyExists($property) && $value === null) {
```

Comment:
As this misses checks for virtual properties this is incomplete. I would expect `getOrFail('full_name')` to work.

Reply:
you are right ;-)

Reply:
Checking virtual properties is done now.